edumanage/forms.py

```diff
@@ -195,6 +197,38 @@ def clean_host(self):
             raise forms.ValidationError(_('This field is required.'))
         return self.cleaned_data["host"]
 
+    def clean_psk_identity(self):
+        psk_identity = self.cleaned_data['psk_identity']
+        if not 'proto' in self.cleaned_data:
+                raise forms.ValidationError(_('The Protocol field is required to validate this field.'))
+        proto = self.cleaned_data['proto']
+        if not 'ertype' in self.cleaned_data:
+                raise forms.ValidationError(_('The Type field is required to validate this field.'))
+        ertype = self.cleaned_data['ertype']
+        if proto in RADPROTOS.TLSPSK:
```

Comment:
Sorry @ghalse , I should have brought this up in my initial review - had it in my notes but missed it when submitting  the review.

This line would check for substring match - I think it should really be: 
```suggestion
        if proto == RADPROTOS.TLSPSK:
``` - as is e.g. in `edumanage/views.py` and `edumanage/management/commands/servdata.py`

I've seen the `proto in RADPROTOS.TLSPSK` pattern here (twice) and also in `djnro/templates/edumanage/server_details.html` - believe it should be `==` instead.

Cheers,
Vlad

Reply:
You're right. I guess this was me stylistically trying to match the existing `if server.ertype in ERTYPE_ROLES.IDP` pattern without thinking through the distinction. Fixed.